langchain/chains/base.py

```diff
@@ -205,11 +209,25 @@ async def acall(
             inputs,
         )
         try:
-            outputs = (
-                await self._acall(inputs, run_manager=run_manager)
-                if new_arg_supported
-                else await self._acall(inputs)
-            )
+            if timeout:
+                # limit execution time for this coroutine
+                outputs = await asyncio.wait_for(
+                    self._acall(inputs, run_manager=run_manager)
+                    if new_arg_supported
+                    else self._acall(inputs),
+                    timeout,
+                )
+            else:
+                # if no timeout specified, use original logic
```

Comment:
nit: the term "original logic" looses meaning after this pr, so comment should be adjusted